src/pages/trainer/training/edit/actions/updateEditor.ts

```diff
@@ -0,0 +1,15 @@
+import {trainerActionEnums} from '../../../../../common/actionEnums/trainer';
+import {textAreaTool} from '../../../../../common/ui/tools/textAreaTool';
+
+export const updateEditorAction = (editor: HTMLTextAreaElement, caret: string, offset: number) => {
+  const contentWithCaret = textAreaTool.insertAtCaretGetText(editor, caret, offset);
+  const cursorStartPosition = textAreaTool.caculateStartCursorPositionPlusOffset(editor, offset);
```

Comment:
So, you mean textAreaTool stuff should be placed in Editor component (retrieve text with carte + calculate start position + set focus) and then when it's all completed, send an action to update content?